modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2897,6 +2897,12 @@ def groupby_agg(
         how="axis_wise",
         drop=False,
     ):
+        # Defaulting to pandas in case of an empty frame
+        if len(self.columns) == 0 or len(self.index) == 0:
+            return super().groupby_agg(
+                by, agg_func, axis, groupby_kwargs, agg_args, agg_kwargs, how, drop
+            )
+
```

Comment:
we don't default to pandas at the front end in case of a lazy query compiler and so there are still cases where empty data could leak into qc

Reply:
We should probably add an inline comment regarding this.